### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by jakbyte**
Just fixed some warnings and satisfied clippy.

### CODE REVIEW

File: src/compositor/mod.rs [hunk 1]
```diff
@@ -57,7 +57,7 @@ impl Compositor {
     /// Poses are relative to the origin set by `set_tracking_space`.
     pub fn wait_get_poses(&self) -> Result<WaitPoses, CompositorError> {
         unsafe {
-            let mut result: WaitPoses = mem::uninitialized();
+            let mut result: WaitPoses = mem::MaybeUninit::uninit().assume_init();
```
Comment (Ralith) [inline src/compositor/mod.rs]: If we're going to adopt `MaybeUninit`, we should so so correctly, i.e. don't call `assume_init` until *after* the memory has been initialized.

File: src/compositor/mod.rs [hunk 2]
```diff
@@ -57,15 +57,19 @@ impl Compositor {
...
+            let mut game: mem::MaybeUninit<TrackedDevicePoses> = mem::MaybeUninit::uninit();
             let e = self.0.WaitGetPoses.unwrap()(
-                result.render.as_mut().as_mut_ptr() as *mut _,
-                result.render.len() as u32,
-                result.game.as_mut().as_mut_ptr() as *mut _,
-                result.game.len() as u32,
+                render.as_mut_ptr() as *mut _,
+                render.assume_init().len() as u32,
+                game.as_mut_ptr() as *mut _,
+                game.assume_init().len() as u32,
```
Comment (Ralith) [inline src/compositor/mod.rs]: Calling `assume_init` before the write has actually taken place is incorrect.

Comment (jakbyte) [reply src/compositor/mod.rs]: Ah, forgot about that. Will fix.

File: src/system/mod.rs [hunk 3]
```diff
@@ -91,14 +101,14 @@ impl System {
...
         unsafe {
-            let mut result: TrackedDevicePoses = mem::uninitialized();
+            let mut result: mem::MaybeUninit<TrackedDevicePoses> = mem::MaybeUninit::uninit();
             self.0.GetDeviceToAbsoluteTrackingPose.unwrap()(
                 origin as sys::ETrackingUniverseOrigin,
                 predicted_seconds_to_photons_from_now,
-                result.as_mut().as_mut_ptr() as *mut _,
-                result.len() as u32,
+                result.as_mut_ptr() as *mut _,
+                result.assume_init().len() as u32,
```
Comment (Ralith) [inline src/system/mod.rs]: Here too.

File: src/lib.rs [hunk 4]
```diff
@@ -188,7 +188,7 @@ unsafe fn get_string<F: FnMut(*mut std::os::raw::c_char, u32) -> u32>(mut f: F)
     }
     let mut storage = Vec::new();
     storage.reserve_exact(n as usize);
-    storage.resize(n as usize, mem::uninitialized());
+    storage.resize(n as usize, mem::MaybeUninit::uninit().assume_init());
```
Comment (Ralith) [inline src/lib.rs]: Here too.

**Pull request closed without merging**